Update custom-collector.md

RichardChukwu opened the pull request:
To resolve the versioning issues, it is necessary to update the builder-config.yaml to include a providers block specifying compatible versions for the envprovider and fileprovider

This change ensures that the correct versions of the providers are referenced, thus preventing errors during the build process.

The doc has been updated to include that. @tiffany76 @radar @jf @dazuma 

Fixes #5259.

tiffany76 commented on content/en/docs/collector/custom-collector.md:
````diff
@@ -164,7 +164,12 @@ processors:
   - gomod:
       go.opentelemetry.io/collector/receiver/otlpreceiver {{% version-from-registry collector-receiver-otlp %}}
-```
+```providers:
````
```suggestion

providers:
```

tiffany76 commented on content/en/docs/collector/custom-collector.md:
```diff
@@ -164,7 +164,12 @@ processors:
+  - gomod: go.opentelemetry.io/collector/confmap/provider/fileprovider v1.16.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/httpprovider v0.110.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/httpsprovider v0.110.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/yamlprovider v0.110.0
```
To close out the yaml block, you'll need to add the triple back tick (```) on line 173 and follow it up with a blank line.

tiffany76 reviewed (changes requested):
Just a couple fixes to the back ticks. Thanks!

tiffany76 commented:
Thanks for the PR, @RichardChukwu! I added the issue number to the description so the issue and PR are linked and others know that you've submitted a fix. :+1:

RichardChukwu commented:
Thanks for the feedback and suggestions @tiffany76  I've implemented all and updated the commit, do let me know if there are any further changes to be made to be ready for merge, thank you

tiffany76 reviewed (approved):
LGTM! @mx-psi, please take a look!

svrnm commented:
>  @svrnm I'd also appreciate feedback and possible approval

since @tiffany76 reviewed and approved mine is not necessary, let's wait for @mx-psi or anyone from @open-telemetry/collector-approvers to take a look as well.

RichardChukwu commented:
Noted @svrnm thank you very much

mx-psi commented on content/en/docs/collector/custom-collector.md:
```diff
@@ -164,6 +164,13 @@ processors:
+
+providers:
+  - gomod: go.opentelemetry.io/collector/confmap/provider/envprovider v1.16.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/fileprovider v1.16.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/httpprovider v0.110.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/httpsprovider v0.110.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/yamlprovider v0.110.0
```
Shouldn't this use the `version-from-registry` thingie? It should work regardless so I am approving (confmap is 1.0 so this will work fine indefinitely now), but wondering if we should be consistent with the above

RichardChukwu replied:
Ook, I get what you mean though, so should I use that instead or this is good to go for merge already?

mx-psi replied:
I am fine to merge like this since it should work fine, and deal with updating this later

RichardChukwu replied:
Alright. Thank you for the feedback.  Much appreciated.

RichardChukwu commented:
> >  @svrnm I'd also appreciate feedback and possible approval
> 
> since @tiffany76 reviewed and approved mine is not necessary, let's wait for @mx-psi or anyone from @open-telemetry/collector-approvers to take a look as well.

Hello @svrnm , @mx-psi has approved, is there any further action for merge please?

Thank you

svrnm commented:
> > > @svrnm I'd also appreciate feedback and possible approval
> > 
> > 
> > since @tiffany76 reviewed and approved mine is not necessary, let's wait for @mx-psi or anyone from @open-telemetry/collector-approvers to take a look as well.
> 
> Hello @svrnm , @mx-psi has approved, is there any further action for merge please?

No, we are good, will merge it now. Please note that working on open source projects is an inherently async process, so it is not unusual that it can take a few days until people get back to things. There is no issue to nudge people after they have not been responsive for a while, but depending on the urgency of your matter, give them at least 24 hours, maybe even 2-3 days to get back to you.

Thanks for putting in the effort! 🚀

RichardChukwu commented:
Thank you very much, duly noted.
